File: all-submissions/0829-subdomain-visit-count/solution.java
class Solution {
    public List<String> subdomainVisits(String[] cpdomains) {
        Map<String, Integer> map = new HashMap();
        
        for(String str : cpdomains){
            String[] line = str.split(" ");
            int count = Integer.valueOf(line[0]);
            String[] domains = line[1].split("\\.");
            String temp = "";
            for(int i = domains.length - 1;i >= 0;i--)
            {
                temp = domains[i] + (temp.equals("") ? temp : "." + temp);
                map.put(temp, map.getOrDefault(temp,0) + count);
            }
        }
        
        List<String> res = new ArrayList();
        for(String str : map.keySet()){
            res.add(map.get(str) + " " + str);
        }
        
        return res;
    }
}

File: all-submissions/0829-subdomain-visit-count/solution.py
from collections import defaultdict
class Solution:
    def subdomainVisits(self, cpdomains: List[str]) -> List[str]:
        domainCountMap = defaultdict(int)
        for i in range(len(cpdomains)):
            count, domainName = cpdomains[i].split()
            subdomains = domainName.split(".")
            for i in range(len(subdomains)):
                d = ".".join(subdomains[i:])
                domainCountMap[d] += int(count)
        
        result = []
        for domain, count in domainCountMap.items():
            result.append(str(count) + " " + domain)
            
        return result
            
        
